pdns/dnsname.hh

```diff
@@ -273,16 +301,48 @@ struct SuffixMatchTree
     }
     return *this;
   }
+  bool operator<(const SuffixMatchTree& rhs) const
+  {
+    return strcasecmp(d_name.c_str(), rhs.d_name.c_str()) < 0;
```

Comment:
`strcasecmp()` is locale dependent, which in theory could lead to surprises. A method in the style of DNSName::canonCompare() is better I think. Time to factor that out?

Reply:
In theory that makes sense to me, yes. In practice we need to be careful because `DNSName::canonCompare()` uses `DNSName::getRawLabels()`, so probably something else is needed.
It also feels a bit out of scope for this PR to me :)

Reply:
I'm worried this code suddenly becomes slow if you do not use the C locale. I'm OK with doing it separately.

Reply:
Yeah, this is a valid concern. I'll open an issue so we can track it.